### PR TITLE
Multiple cleanups

### LINKED ISSUE #225: Allow explicitly naming parameter group, so that Blue Green upgrades can be done across major versions
### Describe the Feature

We would like to be able to explicitly specify the names of the parameter groups created by this module

### Expected Behavior

We need to be able to explicitly set the name of the parameter groups created by this module.

### Use Case

We are performing a Blue/Green upgrade on our Aurora MySQL database, moving from MySQL 5.7-compatible Aurora to MySQL 8.x-compatible Aurora. To do so, we had to create a new 8.x-compatible parameter group to attach to the green cluster and instances. Since the two parameter groups need to exist concurrently, the new parameter group must have a different name than the original one.

Once we have completed the Blue/Green upgrade, however, if we use `terraform state rm` to remove the old parameter group from state, and `terraform state import` to add the new one in its place, the module forces a replacement because the name is incorrect.

### Describe Ideal Solution

The ideal solution would be to be able to explicitly specify the name of the parameter group, overriding the module's calculated name.

### Alternatives Considered

We basically have two alternatives:
1. We can create a fork of the module that allows us to explicitly set the parameter group name.
2. After completing the Blue/Green upgrade, we can remove the original blue parameter group, and then allow terraform to perform a replacement of the new parameter group, creating a new one matching the original naming scheme. Of course, syncing a new parameter group forced a cluster reboot, generating downtime, which somewhat defeats (part of) the purpose of doing a Blue/Green upgrade to begin with.

### Additional Context

_No response_

### LINKED ISSUE #241: 1.13.0 is forcing a recreate of the cluster instance
### Describe the Bug

I'm currently using 1.12.0, which is good and all.

I'd like to upgrade to 1.14.0 which has a bug fix in an output that I'd like to utilize. 

BUT a PR merged for 1.13.0 is causing the DB instance to recreate due to the way the attribute is defined (even if the end result value isn't changing). 

See that PR here https://github.com/cloudposse/terraform-aws-rds-cluster/pull/236#pullrequestreview-2479491627

https://sweetops.slack.com/archives/CB6GHNLG0/p1733335202382489

### Expected Behavior

It should not force a recreate of the resource. If it forces a recreate, it should be a MAJOR bump.

### Steps to Reproduce

Set version to 1.13.0, will force replacement
image

### Screenshots

_No response_

### Environment

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Nuru**
## what

- Partly revert #236 (see discussion below)
- Allow explicitly naming parameter group
- Use full name of engine for reserved DB instances
- Implement and close #197 
- Implement and close #199 
- Implement and close #230 Thank you @raymondchen625 
- Implement and close #237 Thank you @Junaid-Tem 
- Implement and close #239 Thank you @oycyc 

## why

- Closes #241 
- Closes #225 
- Instance engine should match cluster engine
- Consolidate PRs

## discussion: why partially revert #236?

PR #236 enhanced the `random_pet` that determines part of the name of the DB instances so that the name would change whenever the instances would need to be recreated. Unfortunately, as a side-effect, that causes all instances created with earlier versions of this module to be replaced.

Upon further investigation, it was determined that all the "keepers" added by the PR would also force the cluster to be replaced. Unlike replacing an instance, replacing the cluster results in data loss. Also, unless you change the name of the cluster, the module both before and after the PR would fail, because it would try to create a new cluster with the same name before destroying the old cluster. 

We prefer this failure mode, requiring the user to explicitly destroy the cluster before creating the new one, because it puts the user on notice about the potential data loss. So given that the changes in the PR did not make something work that did not work before, and it did cause disruption, we reverted the change to the keepers.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+  instance_class              = var.serverlessv2_scaling_configuration != null ? "db.serverless" : var.instance_type
 
   cluster_instance_count   = local.enabled ? var.cluster_size : 0
   is_regional_cluster      = var.cluster_type == "regional"
   is_serverless            = var.engine_mode == "serverless"
   is_serverless_v2         = var.instance_type == "db.serverless" && (contains(["aurora-postgresql", "aurora-mysql"], var.engine)) && var.engine_mode == "provisioned"
   enable_http_endpoint     = var.enable_http_endpoint && (local.is_serverless || local.is_serverless_v2)
   ignore_admin_credentials = var.replication_source_identifier != "" || var.snapshot_identifier != null
-  reserved_instance_engine = split("-", var.engine)[1]
+  reserved_instance_engine = var.engine
```
Comment (oycyc) [inline main.tf:16]: I haven't used this but will soon through this module. Wondering if this is intended / or what is the rationale for this?

In the examples complete, I'm seeing engine is passed in as `aurora-postgres` or `aurora-mysql`
and in the Terraform docs
image

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_reserved_instance

and https://github.com/cloudposse/terraform-aws-rds-cluster/blob/04c61c2c00cf19c0708e98b5b574ff50b4460a78/main.tf#L105

File: main.tf [hunk 2]
```diff
@@ -108,6 +112,8 @@ resource "aws_rds_reserved_instance" "default" {
 
   offering_id    = data.aws_rds_reserved_instance_offering.default[0].id
   instance_count = local.cluster_instance_count
+  reservation_id = var.rds_ri_reservation_id
```
Comment (oycyc) [inline main.tf:115]: 🙌

**Comment by oycyc** [comment]
Appreciate this! Thanks for maintaining this module @Nuru 🙌

File: main.tf (hunk 1 above)
Comment (Nuru) [reply main.tf:16]: The product description is supposed to match the engine exactly. `aurora-mysql` and `mysql` are separate available engines.

**Comment by oycyc** [comment]
Ahhh I totally forgot about that. Thanks for pointing that out!

File: main.tf (hunk 1 above)
Comment (oycyc) [reply main.tf:16]: Ahhh I totally forgot about that. Thanks for pointing that out!

**Review by korenyoni — APPROVED** [review]
LGTM

**Pull request merged**